Jenkinsfile

```diff
@@ -523,53 +531,74 @@ pipeline {
          *
          * EXECUTION CONDITIONS
          * --------------------
-         * - PIPELINE_CONTROL.ci_skip is false
-         * - PIPELINE_CONTROL.deploy is true
-         * - The build is still successful and not unstable
+         * - Always
          *
-         * DESCRIPTION
-         * -----------
-         * Bumps the pre-release version in preparation for publishing to an npm
-         * registry. 
+         * DECRIPTION
+         * ----------
+         * Compare the version in package.json with released version. If the same,
+         * the build is stopped. Needed because the pipeline does do some simple
+         * merge on the master branch for documentation updates
+         *
+         * OUTPUTS
+         * -------
+         * PIPELINE_CONTROL.cd_skip will be set to 'true', 
+         *  - if master branch's release version and new version are the same
+         *  - if there isn't any new commit
          ************************************************************************/
-        stage('Bump Version') {
+        stage('Check for CD Skip') {
         when {
                 allOf {
                     expression {
                         return PIPELINE_CONTROL.ci_skip == false
                     }
                     expression {
-                        return PIPELINE_CONTROL.deploy
+                        return PIPELINE_CONTROL.cd_skip == false
                     }
                     expression {
                         return RELEASE_BRANCHES.contains(BRANCH_NAME)
                     }
-                    expression {
-                        return GIT_COMMIT != GIT_PREVIOUS_SUCCESSFUL_COMMIT
-                    }
                 }
             }
             steps {
-                timeout(time: 5, unit: 'MINUTES') {
-                    echo "Bumping Version"
-
-                    // This npm command does the version bump
+                timeout(time: 2, unit: 'MINUTES') {
                     script {
-                        def baseVersion = sh returnStdout: true, script: 'node -e "console.log(require(\'./package.json\').version.split(\'-\')[0])"'
-                        def preReleaseVersion = baseVersion.trim() + "-next." + new Date().format("yyyyMMddHHmm", TimeZone.getTimeZone("UTC"))
-                        sh "npm version ${preReleaseVersion} --no-git-tag-version"
+                        // Retrieve new version from package.json
+                        def NEW_VERSION = sh (
+                            script: 'echo "console.log(require(\'./package.json\').version);" | node',
+                            returnStdout: true
+                        ).trim()
+                        echo "NEW_VERSION: ${NEW_VERSION}"
+                        // Retrieve released version from npmjs.org
+                        def RELEASED_VERSION = sh (
+                            script: 'npm view "$(echo "console.log(require(\'./package.json\').name);" | node)" version || true',
```

Comment:
Is it possible to add `--registry $NPM_REGISTRY` to the `npm view` command to make sure we're not picking up anything different from `.npmrc`?

Reply:
done